modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1987,6 +1987,20 @@ void GDScriptAnalyzer::resolve_function_body(GDScriptParser::FunctionNode *p_fun
 	}
 	p_function->resolved_body = true;
 
+	if (p_function->is_abstract) {
+		// Abstract functions don't have a body.
+		if (!p_function->body->statements.is_empty()) {
+			push_error(R"(Abstract function cannot have a body.)", p_function->body);
+		}
+		return;
+	} else {
+		// Non-abstract functions must have a body.
+		if (p_function->body->statements.is_empty()) {
+			push_error(R"(Expected function to either have ":" followed by a function body, or "@abstract" for an abstract function.)", p_function);
```

Comment:
```suggestion
			push_error(R"(A function must either have a ":" followed by a body, or be marked as "@abstract".)", p_function);
```

Not sure if this is better, just trying to eliminate potential confusion with "followed by" due to the annotation having to **precede** the function declaration.